host.py

```diff
@@ -328,8 +328,7 @@ def _run_local(
                     full_env.pop(k, None)
                 else:
                     full_env[k] = v
-        args = shlex.split(cmd)
-        res = subprocess.run(args, capture_output=True, env=full_env)
+        res = subprocess.run(cmd, shell=True, capture_output=True, env=full_env)
```

Comment:
Can we instead make run_remote not use a shell?

Reply:
Yes. But why?

The commit message elaborates on this already.

It's significantly less powerful, but for what benefit?

If you simply don't want to use a shell, then don't. In the majority of cases, we don't need it. But there is no need to take that option away for the cases where it's useful.

I guess, you also don't want an optional argument like `subprocess.run()`'s `shell=$BOOL`. So making it always a non-shell, makes it cumbersome to use a shell. If you are worried about bad style, don't use bad style. There are still ways to use bad style.

For example, I presume you also dislike the many arguments of `subprocess.run()` that are about how to handle stdout/stderr. If we have a shell, then at least we can do `cmd="make deploy 2>&1"` so that the entire output is combined. Which can sometimes be useful to not have the two streams separate and hard to read.

Also, there exist various callers in the code, that make use of a shell already. Those would also need to be updated.

Reply:
I prefer not to allow shells to discourage commits that introduce things like https://github.com/bn222/cluster-deployment-automation/commit/55e67b3a3775d980fd308dd6203aacec4dd620a1#diff-dbbc495980580c8315c8de807876dfd44cd4adaa99dc96ef1f38d063b31f8aa7R48

I prefer to keep this a python project.